lib/dynamoid/transaction_write/action.rb

```diff
@@ -37,6 +41,25 @@ def range_key
         attributes[model_class.range_key]
       end
 
+      # sets a value in the attributes
+      def set(values)
+        attributes.merge!(values)
+      end
+
+      # increments a number or adds to a set, starts at 0 or [] if it doesn't yet exist
+      def add(values)
+        additions.merge!(values)
+      end
+
+      # deletes a value or values from a set type or simply sets a field to nil
+      def delete(values)
+        if values.is_a?(Hash)
+          deletions.merge!(values)
+        else
+          set(values => nil)
```

Comment:
Currently `ItemUpdater#delete` without value means removing the whole attribute in an item, not just its nullifying. In terms of [UpdateExpression](https://docs.aws.amazon.com/amazondynamodb/latest/APIReference/API_UpdateItem.html#API_UpdateItem_RequestSyntax) it's `REMOVE` operation.

Reply:
Indeed. I'll look into this. My understanding is REMOVE is functionally the same as setting to nil so there's no way to write a test distinguishing them.